### ISSUE TITLE
Retrieve identification software scores for PSMs

### ISSUE THREAD

**Issue by videlc** [compomics/peptide-shaker#273]
Hi,

I am looking to extract different softwares scores from a quite large number of files. I know this information is stored in cpsx as well as mzID when this output is created. However, I can't find a way to retrieve this information. Ideal would be PSMs information + 1 column per identification algorithm with score for each (as well as peptideshaker PSM score).

My initial idea was to import peptideshaker-mzID into mzR but this doesn't show score either.

Also a good point would be to be able to output the information about the agreement of software algorithm (if all of the software or some agreed on the identification of each PSM).

Is there a way to do this ?

Best regards,
Vivian

**Comment by hbarsnes** [comment]
Hi Vivian,

Have you looked into the Identification Feature exports? You'll find them via Export > Identification Features in the graphical user interface or via the [ReportCLI command line](https://github.com/compomics/peptide-shaker/wiki/PeptideShakerCLI#b---reportcli).

You can make your own custom reports if the default reports do not cover your needs.

Best regards,
Harald

**Comment by videlc** [comment]
Hi Harald,

Thanks for pointing me there ! This solved my problem partially as now I am able to extract wanted information _via_ graphic interface. However, as I'm dealing with a lot of files, I'd like to setup a command-line version of the wanted custom report. 

Could you show me the correct synthax to set it ? 
In your wiki, it is written  `-reports "9-n"` will give custom report. However, neither 9 nor "-" are accepted by ReportCLI 

```
 -reports "9-3"
<CompomicsError>PeptideShaker processing failed. See the PeptideShaker log for details.</CompomicsError>java.lang.IllegalArgumentException: Cannot parse 9-3 into an integer from 9-3 with separator ','.
        at com.compomics.software.cli.CommandLineUtils.getIntegerListFromString(CommandLineUtils.java:235)
        at eu.isas.peptideshaker.cmd.ReportCLIInputBean.<init>(ReportCLIInputBean.java:63)
        at eu.isas.peptideshaker.cmd.ReportCLI.main(ReportCLI.java:348)
```

Let's say I want a custom report subtype number 3 with columns 1,2,3,4,5,15,30, how should I build the command ?

Best regards,
Vivian

**Comment by hbarsnes** [comment]
Hi Vivian,

The number following the -reports option is the number of the report you want to export. For example -reports "0, 3" would give you report type 0 which is "Certificate of Analysis" and report 3 which is "Default PSM Report".

9-n refers to the number of your custom reports created in the graphical user interface. The first will be number 9, the second 10, etc. Running the ReportCLI command line without any parameters, i.e. "java -cp PeptideShaker-X.Y.Z.jar eu.isas.peptideshaker.cmd.ReportCLI" will give you the overview of the number connected to each custom report.

In other words, if you want to export the custom report listed as report number nine (which you previously have set up in the graphical user interface to contain the columns you need), you use the option "-reports 9".

Best regards,
Harald

**Comment by videlc** [comment]
Thanks Harald.

Last things keeping me from doing it is that I created that custom report on my personal computer and I need to run this on a PC cluster. Can you tell me where to find that report parameter file on my personal computer ? 

I guess I'd have to transfer this parameter file to peptideshaker path on cluster and then I'm good to go, right ?

Best,
Vivian

**Comment by hbarsnes** [comment]
Hi Vivian,

The reports are stored in the file called exportFactory.cus, which is by default located in the .peptideshaker folder in your user home folder. You can see the actual location of the file on your computer via the PeptideShaker Welcome dialog > Settings & Help > Settings > Resource Settings. It is the one labeled "peptideshaker_exports". Simply replace the exportFactory.cus file on your cluster with the one from your personal computer and the custom reports should be available.

Best regards,
Harald

**Comment by videlc** [comment]
Thanks Harald,

Will give it a try today !
One question is what are these values :
`OMSSA (7.83101082312125E-10), X!Tandem (0.05), MS-GF+ (1.2168469E-18), Comet (1.11E-4), MyriMatch (73.69879080715)`

Are they scores/evalues/PEP. 
Are they directly correlated with scores ? (A log10 transformation of the xE-X would be well correlated with each algorithm scores ?

Vivian

**Comment by hbarsnes** [comment]
Hi Vivian,

> Are they scores/evalues/PEP....? Are they directly correlated with scores ?

These are the scores given by the search engines themselves. You will have to refer to the documentation for each search engine to verify what each score represent for the different search engines.

Best regards,
Harald

**Comment by videlc** [comment]
Hi Harald,

Running it on cluster works sometimes and sometimes doesn't.
Success rate is 1584/4977.

Working : 

```
An error occurred when setting path configurations. Default paths will be used.
10% 20% 80% 90%Reindexing: searchgui_db_concatenated_target_decoy.fasta.

10% 20% 30% 40% 50% 60% 70% 80% 90%
10% 20% 30% 40% 50% 60% 70% 80% 90%
Sat Jul 08 06:53:57 EDT 2017 Exporting. Please Wait...
10% 20% 30% 40% 50% 60% 70% 80% 90% 100%
Sat Jul 08 06:55:12 EDT 2017 Report export completed.
```


Not working : 
```
An error occurred when setting path configurations. Default paths will be used.
10% 70% 80% 90%Reindexing: searchgui_db_concatenated_target_decoy.fasta.

10% 20% 30% 40% 50% 60% 70% 80% 90%
10% 20% 30% 40% 50% 60% 70% 80% 90%
Sat Jul 08 05:14:54 EDT 2017 Exporting. Please Wait...
Sat Jul 08 05:14:54 EDT 2017 An error occurred while exporting the test6.
```

Any idea where that might coming from ?

Best regards,
Vivian

**Comment by hbarsnes** [comment]
Hi Vivian,

> Running it on cluster works sometimes and sometimes doesn't. Any idea where that might coming from ?

Are there any details about the errors that you can provide? There should be a log file containing the Java stack traces, depending on how you set up the path configuration. And you should look into the error with the setting of the path configurations btw. Could be related.

Best regards,
Harald

**Comment by videlc** [comment]
Hi Harald,

As an update, we processed smaller batchs of samples. We did numerous modifications (such as using more recent versions of PeptideShaker) but the fix came from one specific change.

In previous tests, we were calling PeptideShaker jar out of the sample folder for every sample. 

This resulted in many errors as follow : 
`Tue Jul 11 12:59:30 EDT 2017 An error occurred while reading: file.cpsx. It looks like another instance of PeptideShaker is still connected to the file. Please close all instances of PeptideShaker and try again.`

Copying the jar on each used computer node as the first step and running peptideshaker commands from that place solved the issue. Will now process more samples. Will update if issues are faced.

Thanks again for your help !
Vivian

**Comment by hbarsnes** [comment]
Hi Vivian,

Great! I will then close the issue. But do not hesitate to open a new one if you come across other problems.

Best regards,
Harald
